hadoop-project/pom.xml

```diff
@@ -93,11 +93,11 @@
     <!-- Protobuf scope in other modules which explicitly import the libarary -->
     <transient.protobuf2.scope>${common.protobuf2.scope}</transient.protobuf2.scope>
     <!-- ProtocolBuffer version, actually used in Hadoop -->
-    <hadoop.protobuf.version>3.21.12</hadoop.protobuf.version>
+    <hadoop.protobuf.version>3.23.4</hadoop.protobuf.version>
```

Comment:
I'd prefer editing hadoop-thirdparty.version

Reply:
this value is the one used by the protoc compilation and the build fails if I don't upgrade it

Reply:
this is correct, you need to update the version for the protoc generation, thirdparty protobuf isn't used for generating the protoc classes. We generate the classes & then do a replacement in those genrated classes to use the shaded protobuf

https://github.com/apache/hadoop/blob/10ab8abccd7e086d1d47eb7d706c8d5c42052172/hadoop-project/pom.xml#L2269-L2292